server/src/main/java/io/horizen/services/impl/NscServiceImpl.java

```diff
@@ -0,0 +1,192 @@
+package io.horizen.services.impl;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.inject.Inject;
+import io.horizen.config.Settings;
+import io.horizen.helpers.Helper;
+import io.horizen.helpers.MyGsonManager;
+import io.horizen.services.NscService;
+import org.web3j.abi.TypeDecoder;
+import org.web3j.abi.datatypes.Address;
+import org.web3j.abi.datatypes.generated.Bytes3;
+import org.web3j.abi.datatypes.generated.Bytes32;
+import org.web3j.abi.datatypes.generated.Uint32;
+import org.web3j.crypto.Hash;
+import org.web3j.crypto.Keys;
+import org.web3j.utils.Numeric;
+import spark.utils.IOUtils;
+
+import java.net.HttpURLConnection;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.regex.Pattern;
+
+public class NscServiceImpl implements NscService {
+    private final Settings settings;
+
+    @Inject
+    public NscServiceImpl(Settings settings) {
+        this.settings = settings;
+    }
+
+    public Map<String, List<String>> getNscOwnerships(String scAddress) throws Exception {
+        String method;
+        String abiString;
+        if (scAddress == null) {
```

Comment:
We should not support this case since snapshot dapp should not call it

Reply:
Done